### PR TITLE
Send change event on date picker input change

### PR DESCRIPTION
**Pull request by rudolpho1**
Extended the date picker code to send change events when the user edits the input field.

I had the problem that I did not get change events from the date picker if the user just typed something into the input field (not using the picker) and then left the field without pressing enter, e.g. by pressing tab or clicking somewhere else on the screen. In this case my component did not get a change event from the date picker.

#### Changelog

**New**

* added an output property "valueChange" to the datepicker-input component

* extended the datepicker component to listen to "valueChange" from datepicker input and to change the flatpickr value accordingly

### CODE REVIEW

File: src/datepicker/datepicker.component.ts [hunk 1]
```diff
@@ -127,4 +128,32 @@ export class DatePicker {
 			}
 		});
 	}
+
+	onInputValueChange(event: string): void {
+		const d = flatpickr.parseDate(event, this.dateFormat, true);
```
Comment (magedhennawy) [inline src/datepicker/datepicker.component.ts]: Hey, we prefer using full words for our variables, maybe something like:

```suggestion
		const eventDates = flatpickr.parseDate(event, this.dateFormat, true);
```

File: src/datepicker/datepicker.component.ts [hunk 2]
```diff
@@ -127,4 +128,32 @@ export class DatePicker {
...
+			}
+		} else {
+			if (prev || event) {
+				this.value = [undefined, this.value[1]];
+			}
+		}
+	}
+
+	onInputValueRangeChange(event: string): void {
+		const d = flatpickr.parseDate(event, this.dateFormat, true);
```
Comment (magedhennawy) [inline src/datepicker/datepicker.component.ts]: Same as comment above

**Comment by zvonimirfras** [comment]
There are some conflicts in `src/datepicker/datepicker.component.ts`. Do you mind resolving those before we merge this?

File: src/datepicker/datepicker.component.ts [hunk 3]
```diff
@@ -149,6 +149,34 @@ export class DatePicker implements OnDestroy {
 		});
 	}
 
+	onInputValueChange(event: string): void {
```
[the remark this answers is not in the document]
Comment (rudolpho1) [reply src/datepicker/datepicker.component.ts]: Done.

[the remark this answers is not in the document]
Comment (zvonimirfras) [reply src/datepicker/datepicker.component.ts]: That was awesome!

GitHub is showing me that there's one more suggested change pending, is that a GitHub fluke or real?

That one is to conform to our (still internal but soon to be published) code style guide which states: "Generally imports should be given one line per entry, unless there are 3 or fewer entries."

[the remark this answers is not in the document]
Comment (rudolpho1) [reply src/datepicker/datepicker.component.ts]: Sorry, I somehow oversaw that. It's done now.

**Pull request merged**